### PR TITLE
[hail] Implement concordance in Python.

### PR DESCRIPTION
**Pull request by tpoterba**
This is unfortunately about 2x slower -- partly due to the fact
that the column + global concordance calculations are not fused,
and partly because the AggArrayPerElement stuff seems pretty
slow right now and is dragging down the per-sample concordance.

### CODE REVIEW

File: hail/python/hail/methods/qc.py [hunk 1]
```diff
@@ -285,9 +285,51 @@ def has_field_of_type(name, dtype):
     return mt.annotate_rows(**{name: result})
 
 
+def concordance2(left, right, *, _localize_global_statistics=True):
+    print('conc2')
```
Comment (patrick-schultz) [inline hail/python/hail/methods/qc.py]: The print is clearly debugging code. Is `concordance2` itself also?

Comment (tpoterba) [reply hail/python/hail/methods/qc.py]: whoops! Yes, the whole function can be removed.

File: hail/python/hail/methods/qc.py [hunk 2]
```diff
@@ -402,20 +403,66 @@ def concordance(left, right) -> Tuple[List[List[int]], Table, Table]:
...
+
+    info(f"concordance: including {included} shared samples "
+         f"({len(left_sample_counter)} total on left, {len(right_sample_counter)} total on right)")
+
+    left = require_biallelic(left, 'concordance, left')
+    right = require_biallelic(right, 'concordance, right')
+
+    lit = hl.literal(included, dtype=hl.tset(hl.tstr))
+    left = left.filter_cols(lit.contains(left.col_key[0]))
+    right = right.filter_cols(lit.contains(right.col_key[0]))
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply hail/python/hail/methods/qc.py]: semi join can't do an ordered merge -- we don't store the columns ordered

[the remark this answers is not in the document]
Comment (patrick-schultz) [reply hail/python/hail/methods/qc.py]: Oh of course. Never mind!

**Pull request merged**